src/pages/Compare/index.tsx

```diff
@@ -112,6 +115,12 @@ export class CompareBase extends React.Component<Props> {
           path: path || undefined,
         }),
       );
+      return;
+    }
+
+    if (version && !currentVersionId) {
+      dispatch(versionsActions.setCurrentVersionId({ versionId: version.id }));
+      return;
```

Comment:
This also should not return because `loadData()` may need to [fetch a file](https://github.com/mozilla/addons-code-manager/blob/5b757e1f6084fbab36891c8edb9b7ec2c88149ac/src/pages/Compare/index.tsx#L128).

In case you're wondering why `Compare` and `Browse` seem to have duplicated logic, it's just hard to share this kind of `componentDidMount()` logic in classic React. Once we can [port to hooks](https://github.com/mozilla/addons-code-manager/issues/363) (currently blocked on Enzyme), we can remove some of the duplication.

Reply:
Thanks for letting me know this :)